zfs/zfs.go

```diff
@@ -81,64 +91,10 @@ func (z *zfs) Recv(fs string, sendCommand *exec.Cmd) error {
 
 	return nil
 }
-func (z *zfs) SendIncremental(fs string, prev, snap string) *exec.Cmd {
-	panic("not implemented")
-	return nil
-}
-
-type remoteZfs struct {
-	host string
-	user string
-}
-
-func NewRemote(host string, user string) Zfs {
-	return &remoteZfs{
-		host: host,
-		user: user,
-	}
-}
-
-func (z *remoteZfs) List() Fs {
-	dialstring := fmt.Sprintf("%s@%s", z.user, z.host)
-	cmd := exec.Command("/usr/bin/ssh", dialstring, "/sbin/zfs", "list", "-t", "all", "-o", "name")
-	b, _ := cmd.Output()
-	s := string(b)
-	scanner := bufio.NewScanner(strings.NewReader(s))
-	var f Fs
-	scanner.Scan() // Discarding first row as it is the name of the column
-	for scanner.Scan() {
-		l := scanner.Text()
-		isSnap := strings.Contains(l, "@")
-		if f == nil {
-			if isSnap {
-				panic("First element should not be snapshot. Error.")
-			}
-
-			f = NewFs(z, l)
-		} else {
-			if isSnap {
-				f.AddSnapshot(l)
-			} else {
-				f.AddChild(l)
-			}
-		}
-
-	}
-
-	return f
-}
-
-func (z *remoteZfs) Recv(fs string, sendCommand *exec.Cmd) error {
-	panic("not implemented")
-	return errors.New("e")
 
-}
-func (z *remoteZfs) SendIncremental(fs string, prev, snap string) *exec.Cmd {
-	dialstring := fmt.Sprintf("%s@%s", z.user, z.host)
+func (z *Zfs) SendIncremental(fs string, prev, snap string) *exec.Cmd {
 	snapfqn := fmt.Sprintf("%s@%s", fs, snap)
-	cmd := exec.Command("/usr/bin/ssh", dialstring, "-C", "/sbin/zfs", "send", "-i", prev, snapfqn)
-
-	return cmd
+	return z.exec("-C", "/sbin/zfs", "send", "-i", prev, snapfqn)
```

Comment:
I think the -C  (that enables ssh compression) should be in zfs/cmd.go at line 13

Reply:
Ok, I added the `-C` flag and improved the error handling.

Reply:
The `-C` Should be removed here as you added it in cmd.go ;-)